Document new Prometheus exporter in Console (#4117)
timescale/docs: _partials/_prometheus-integrate.md

[Prometheus][prometheus] is an open-source monitoring system with a dimensional data model, flexible query language, and a modern alerting approach.

This page shows you how to export your $SERVICE_SHORT telemetry to Prometheus:

- For $CLOUD_LONG, using a dedicated Prometheus exporter in $CONSOLE. 
- For $SELF_LONG, using [PostgreSQL Exporter][postgresql-exporter]. PostgreSQL Exporter exposes user-defined 

## Prerequisites

To follow the steps on this page:

- [Download and run Prometheus][install-prometheus].

> - For $CLOUD_LONG, create a target [$SERVICE_LONG][create-service] with the time-series and analytics capability enabled.

Reviewer: ```suggestion
- For $CLOUD_LONG:

      Create a target [$SERVICE_LONG][create-service] with the time-series and analytics capability enabled.
```

## Export $SERVICE_SHORT telemetry to Prometheus

To export your data, do the following:

<Tabs label="Export metrics to Prometheus">

<Tab title="Timescale Cloud">

<Procedure>

To export metrics from a $SERVICE_LONG, you create a dedicated Prometheus exporter in $CONSOLE, attach it to your $SERVICE_SHORT, then configure Prometheus to scrape metrics using the exposed URL. The Prometheus exporter exposes the metrics related to the $SERVICE_LONG like CPU, memory, and storage. To scrape other metrics, use PostgreSQL Exporter as described for $SELF_LONG. The Prometheus exporter is available for [Scale and Enterprise][pricing-plan-features] pricing plans.

1. **Create a Prometheus exporter**

   1. In [$CONSOLE][open-console], click `Exporters` > `+ New exporter`.

   1. Select `Metrics` for data type and `Prometheus` for provider.

      ![Create a Prometheus exporter in Timescale Cloud](https://assets.timescale.com/docs/images/timescale-create-prometheus-exporter.png)

   1. Choose the region for the exporter. Only $SERVICE_SHORTs in the same project and region can be attached to this exporter.
   
   1. Name your exporter. 

>    1. Change the auto-generated Prometheus credentials, if needed. See [official documentation][prometheus-authentication] on basic authentication in Prometheus. 

Reviewer: why if needed? Can you move the see to a new line please.

Author: Because they are suggested for you - you can change them if you want

      ![Prometheus exporter credentials](https://assets.timescale.com/docs/images/prometheus-authentication.png)

1. **Attach the exporter to a $SERVICE_SHORT**

   1. Select a $SERVICE_SHORT, then click `Operations` > `Exporters`. 

   1. Select the exporter in the drop-down, then click `Attach exporter`.

      ![Attach a Prometheus exporter to a Timescale Cloud service](https://assets.timescale.com/docs/images/attach-prometheus-exporter-timescale-cloud.png)

>    The exporter is now attached to your $SERVICE_SHORT. You can unattach it by clicking the trash bin icon in the exporter list. 

Reviewer: ```suggestion
   The exporter is now attached to your $SERVICE_SHORT. To unattach it, click the trash icon in the exporter list. 
```

      ![Unattach a Prometheus exporter from a Timescale Cloud service](https://assets.timescale.com/docs/images/unattach-prometheus-exporter-timescale-service.png)

1. **Configure the Prometheus scrape target**

>    1. Select your service > `Operations` > `Exporters` and click the information icon next to the exporter. You see the exporter details. 

Reviewer: ```suggestion
   1. Select your service, then click `Operations` > `Exporters` and click the information icon next to the exporter. You see the exporter details. 
```

      ![Prometheus exporter details in Timescale Cloud](https://assets.timescale.com/docs/images/prometheus-exporter-details.png)

   1. Copy the exporter URL. 

   1. In your Prometheus installation, update `prometheus.yml` to point to the exporter URL as a scrape target:

      ```yml
      scrape_configs:
       - job_name: "timescaledb-exporter"
         scheme: https
         static_configs:
           - targets: ["my-exporter-url"]
         basic_auth:
           username: "user"
           password: "pass"
      ```
   

>       See [Prometheus documentation][scrape-targets] for details on configuring scrape targets. 

Reviewer: ```suggestion
      See the [Prometheus documentation][scrape-targets] for details on configuring scrape targets. 
```

      You can now monitor your $SERVICE_SHORT metrics. Use the following metrics to check the service is running correctly:

      *   `timescale.cloud.system.cpu.usage.millicores`
      *   `timescale.cloud.system.cpu.total.millicores`
      *   `timescale.cloud.system.memory.usage.bytes`
      *   `timescale.cloud.system.memory.total.bytes`
      *   `timescale.cloud.system.disk.usage.bytes`
      *   `timescale.cloud.system.disk.total.bytes`

      Additionally, use the following tags to filter your results.

      |Tag|Example variable| Description                |
      |-|-|----------------------------|
      |`host`|`us-east-1.timescale.cloud`|                            |
      |`project-id`||                            |
      |`service-id`||                            |
      |`region`|`us-east-1`| AWS region                 |
      |`role`|`replica` or `primary`| For $SERVICE_SHORT with replicas |
   
</Procedure>

</Tab>

> <Tab title="Self-hosted TimescaleDB">

Reviewer 2: Strictly speaking it's also applicable for a cloud prometheus, but customer managed prometheus exporter
